File: app/src/main/java/com/v/vlog/MainActivity.java
package com.v.vlog;

import android.Manifest;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.v.log.LogConfig;
import com.v.log.VLog;

import java.util.List;

import pub.devrel.easypermissions.EasyPermissions;

public class MainActivity extends AppCompatActivity implements EasyPermissions.PermissionCallbacks,
        EasyPermissions.RationaleCallbacks {

    private static final int RC_STORAGE_PERM = 123;
    private static final String[] WRITE_AND_READ_STORAGE =
            {Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //初始化日志系统
        VLog.init(new LogConfig(this));
    }

    public void write2file(View view) {
        write2file();
    }

    public void write2file() {
        if (hasWriteAndReadPermissions()) {

            StringBuffer sb = new StringBuffer();
            for (int i = 0; i < 10; i++) {
                sb.append("压夜费第一反应神樱大祓:" + i);
            }
            VLog.i("PRETTY_LOGGER", sb.toString());
            Toast.makeText(MainActivity.this, "success", Toast.LENGTH_LONG).show();
        } else {
            // Ask for both permissions
            EasyPermissions.requestPermissions(
                    MainActivity.this,
                    "申请存储卡选项用于存储",
                    RC_STORAGE_PERM,
                    WRITE_AND_READ_STORAGE);
        }
    }

    private boolean hasWriteAndReadPermissions() {
        return EasyPermissions.hasPermissions(this, WRITE_AND_READ_STORAGE);
    }

    public void flush(View view) {
        VLog.flush();
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        EasyPermissions.onRequestPermissionsResult(requestCode, permissions, grantResults, this);
    }

    @Override
    public void onPermissionsGranted(int requestCode, @NonNull List<String> perms) {

    }

    @Override
    public void onPermissionsDenied(int requestCode, @NonNull List<String> perms) {

    }

    @Override
    public void onRationaleAccepted(int requestCode) {

    }

    @Override
    public void onRationaleDenied(int requestCode) {

    }

}

File: app/src/main/java/com/v/vlog/MainActivity.kt
package com.v.vlog

import android.Manifest
import androidx.appcompat.app.AppCompatActivity
import pub.devrel.easypermissions.EasyPermissions.PermissionCallbacks
import pub.devrel.easypermissions.EasyPermissions.RationaleCallbacks
import android.os.Bundle
import android.view.View
import com.v.vlog.R
import com.v.log.VLog
import com.v.log.LogConfig
import pub.devrel.easypermissions.EasyPermissions
import com.v.vlog.MainActivity
import android.widget.Toast
import com.v.log.util.*

class MainActivity : AppCompatActivity(), PermissionCallbacks, RationaleCallbacks {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //初始化日志系统
        VLog.init(LogConfig(this, true))
        EasyPermissions.requestPermissions(
            this@MainActivity,
            "申请存储卡选项用于存储",
            RC_STORAGE_PERM,
            *WRITE_AND_READ_STORAGE
        )
    }

    fun write2file(view: View?) {
        if (hasWriteAndReadPermissions()) {
            val sb = StringBuffer()
            for (i in 0..9) {
                sb.append("测试11111111111111111111111111:$i")
            }
            sb.logI()
            Toast.makeText(this@MainActivity, "success", Toast.LENGTH_LONG).show()
        }
    }

    fun writeSave(view: View?) {
        if (hasWriteAndReadPermissions()) {
            val sb = StringBuffer()
            for (i in 0..9) {
                sb.append("测试22222222222222222222222$i")
            }
            sb.logI(save = false)
            Toast.makeText(this@MainActivity, "success", Toast.LENGTH_LONG).show()
        }
    }

    private fun hasWriteAndReadPermissions(): Boolean {
        return EasyPermissions.hasPermissions(this, *WRITE_AND_READ_STORAGE)
    }

    fun flush(view: View?) {
        VLog.flush()
    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<String>,
        grantResults: IntArray
    ) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        EasyPermissions.onRequestPermissionsResult(requestCode, permissions, grantResults, this)
    }

    override fun onPermissionsGranted(requestCode: Int, perms: List<String>) {}
    override fun onPermissionsDenied(requestCode: Int, perms: List<String>) {}
    override fun onRationaleAccepted(requestCode: Int) {}
    override fun onRationaleDenied(requestCode: Int) {}

    companion object {
        private const val RC_STORAGE_PERM = 123
        private val WRITE_AND_READ_STORAGE = arrayOf(
            Manifest.permission.WRITE_EXTERNAL_STORAGE,
            Manifest.permission.READ_EXTERNAL_STORAGE
        )
    }
}